### PR TITLE
grilo: add darwin support

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

As in the title.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/grilo/default.nix [hunk 1]
```diff
@@ -34,6 +34,11 @@ stdenv.mkDerivation rec {
     sha256 = "81Ks9zZlZpk0JwY2/t5mtS2mgB/iD2OMQEirJnhXey0=";
   };
 
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace src/meson.build \
```
Comment (jtojnar) [inline pkgs/development/libraries/grilo/default.nix:38]: Could you please upstream this, see e.g. https://gitlab.gnome.org/GNOME/babl/-/merge_requests/26

**Comment by risicle** [comment]
`nixpkgs-review` reports no new failures for macos 10.15 & nixos x86_64.

Note though that `grilo-plugins` and `gupnp-tools` still report themselves as `broken` for me on macos 10.15.

File: pkgs/development/libraries/grilo/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/libraries/grilo/default.nix:38]: https://gitlab.gnome.org/GNOME/grilo/-/merge_requests/96

**Comment by wegank** [comment]
Oh, building `grilo-plugins` actually requires #196577 to be merged first.

**Comment by wegank** [comment]
@ofborg build grilo

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 196850` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>dino</li>
    <li>gupnp-tools</li>
    <li>mate.caja-extensions</li>
    <li>nheko</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>dleyna-connector-dbus</li>
    <li>dleyna-core</li>
    <li>dleyna-renderer</li>
    <li>dleyna-server</li>
    <li>gom</li>
    <li>grilo</li>
    <li>grilo-plugins</li>
    <li>gssdp</li>
    <li>gupnp</li>
    <li>gupnp-av</li>
    <li>gupnp-dlna</li>
    <li>gupnp-igd</li>
    <li>libmediaart</li>
  </ul>
</details>

File: pkgs/development/libraries/grilo/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/libraries/grilo/default.nix]: Upstreamed.

File: pkgs/development/libraries/gssdp/default.nix [hunk 2]
```diff
@@ -50,7 +50,8 @@ stdenv.mkDerivation rec {
     "-Dintrospection=${lib.boolToString (stdenv.buildPlatform == stdenv.hostPlatform)}"
   ];
 
-  doCheck = true;
+  # Bail out! GLib-GIO-FATAL-CRITICAL: g_inet_address_to_string: assertion 'G_IS_INET_ADDRESS (address)' failed
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gssdp/default.nix:53]: Is this maybe related to local Networking?

Comment (wegank) [reply pkgs/development/libraries/gssdp/default.nix:53]: Yes, it even fails with `__darwinAllowLocalNetworking = true;`.
